### PR TITLE
Microsoft Intune support

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Microsoft Intune support

# Impacts
Provisioner

# Delete branch after merge
YES

# Checklist
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries
## New Features
* Microsoft Intune support

### CODE REVIEW

File: docs/provisioner/intune.asciidoc [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// to display images directly on GitHub
+ifdef::env-github[]
+:imagesdir: ../images
+endif::[]
+
+Configure from the Azure portal
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+You will first need to connect to the Azure portal and be sure that you have the intune licences.
```
Comment (extrafu) [inline docs/provisioner/intune.asciidoc]: the Intune

File: docs/provisioner/intune.asciidoc [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+Creating the application
+++++++++++++++++++++++++ 
+Once you are logged in the portal you need to create a application to allow the access to the Graph API.
```
Comment (extrafu) [inline docs/provisioner/intune.asciidoc]: an application

File: docs/provisioner/intune.asciidoc [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+* The Provisioning ID is the friendly name of the provisioner.
+* The Application ID is the 'Application (client ID)'.
+* The Application Secret is the 'Client secret'.
+* The Tenant ID is the 'Directory (tenant ID)'.
+* The Client Secret is the secret of the application you created in the developer account.
+* The default host should work.
+* The default Login URL should work. 
+* The port and protocol should be left to default.
+* The 'Agent download uri' should be ok.
+* Authorized domains needs to be adapted to allow the device to reach the download uri (per example google play needs multiples domain to be able to install the agent).
```
Comment (extrafu) [inline docs/provisioner/intune.asciidoc]: needs -> need
uri -> URI
google play -> Google Play
multiples domain -> multiple domains

File: html/pfappserver/lib/pfappserver/Form/Config/Provisioning/intune.pm [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+has_field 'domains' =>
+  (
+   type => 'Text',
+   label => 'Authorized domains',
+   required => 1,
+   default => 'play.google.com,portal.manage.microsoft.com,apps.apple.com,docs.microsoft.com',
+   element_attr => {'placeholder' => 'play.google.com,portal.manage.microsoft.com,apps.apple.com,docs.microsoft.com'},
+   element_class => ['input-xlarge'],
+   tags => { after_element => \&help,
+             help => 'Comma separated list of domains that will be resolve with the correct IP addresses.' },
```
Comment (extrafu) [inline html/pfappserver/lib/pfappserver/Form/Config/Provisioning/intune.pm]: Comma-separated
be resolve -> be resolved

**Comment by extrafu** [comment]
@julsemaan Please test and @jrouzierinverse please review the code.

File: docs/provisioner/intune.asciidoc [hunk 5]
```diff
@@ -0,0 +1,73 @@
+// to display images directly on GitHub
+ifdef::env-github[]
+:imagesdir: ../images
+endif::[]
+
+Configure from the Azure portal
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+You will first need to connect to the Azure portal and be sure that you have the Intune licences.
```
Comment (jrouzierinverse) [inline docs/provisioner/intune.asciidoc]: Spelling:
licences -> licenses

File: docs/provisioner/intune.asciidoc [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+On the next page you will be able to configure the application, first copy the 'Application (client ID)' and the 'Directory (tenant ID)', you will need them to define your provisioner.
+
+image::Azure_Application_3.png[scaledwidth="100%",alt="Azure ID"]
+
+Next click on 'Certificates & secrets' and 'New client secret' , this will provide you the password to use for the application (Save it right now because you won't be able to have it after).
+
+image::Azure_Application_4.png[scaledwidth="100%",alt="Azure Password"]
+
+Last thing, you need to add permissions on the API, to do that click on 'API permissions' and 'Microsoft Graph' then on the right pane select 'Application permissions' and add:
```
Comment (jrouzierinverse) [inline docs/provisioner/intune.asciidoc]: Reword:
Last -> The last

File: docs/provisioner/intune.asciidoc [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+* The Provisioning ID is the friendly name of the provisioner.
+* The Application ID is the 'Application (client ID)'.
+* The Application Secret is the 'Client secret'.
+* The Tenant ID is the 'Directory (tenant ID)'.
+* The Client Secret is the secret of the application you created in the developer account.
+* The default host should work.
+* The default Login URL should work. 
+* The port and protocol should be left to default.
+* The 'Agent download uri' should be ok.
```
Comment (jrouzierinverse) [inline docs/provisioner/intune.asciidoc]: uri -> URI

File: docs/provisioner/intune.asciidoc [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+Now that you have created the provisioner, go in the 'Connection Profiles' menu on the left and select the default portal.
+Click 'Add Provisioner' and select the new Microsoft Intune provisioner that was created earlier.
+
+image::intune-portal.png[scaledwidth="100%",alt="OPSWAT portal configuration"]
+
+Testing
+^^^^^^^
+You can now test that the installation of the OPSWAT Metadefender Endpoint client is mandatory after the device registration.
+Connect a device to your test network and register like you normally would.
+At the end of the registration process you will be presented a page asking you to install the Intune client on your device.
```
Comment (jrouzierinverse) [inline docs/provisioner/intune.asciidoc]: Add a comma after 'At the end of the registration process'

File: docs/provisioner/intune.asciidoc [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+After you install the client click continue. If your access is enabled than this means the connectivity between PacketFence and Azure is good.
```
Comment (jrouzierinverse) [inline docs/provisioner/intune.asciidoc]: Reword:
than -> then

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationProvisionings.js [hunk 10]
```diff
@@ -715,6 +806,20 @@ export const pfConfigurationProvisioningFields = {
         }
       ]
     }
+  },
+  domains: ({ options: { meta = {} } } = {}) => {
+    return {
+      label: i18n.t('Authorized domains'),
+      text: i18n.t('Comma-separated list of domains that will be resolve with the correct IP addresses.'),
```
Comment (jrouzierinverse) [inline html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationProvisionings.js]: Reword:
A comma-separated list of domains that will be resolved with the correct IP addresses.

**Pull request merged**